instrumentation/opentelemetry-instrumentation-aiohttp-server/src/opentelemetry/instrumentation/aiohttp_server/__init__.py

```diff
@@ -0,0 +1,174 @@
+import functools
+import typing as t
+
+from aiohttp.web import BaseRequest, RequestHandler, StreamResponse
+
+from opentelemetry import context, propagate, trace
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.instrumentation.utils import (
+    extract_attributes_from_object,
+    http_status_to_status_code,
+)
+from opentelemetry.semconv.trace import SpanAttributes
+from opentelemetry.util.http import (
+    ExcludeList,
+    get_excluded_urls,
+    get_traced_request_attrs,
+)
+
+from .version import __version__
+
+RequestHook = t.Callable[[trace.Span, BaseRequest], t.Any]
+ResponseHook = t.Callable[[trace.Span, BaseRequest, StreamResponse], t.Any]
+
+
+_CONFIG_PREFIX = "AIOHTTP_SERVER"
+_CONFIG_TRACED_REQUEST_ATTRS = get_traced_request_attrs(_CONFIG_PREFIX)
+_CONFIG_EXCLUDED_URLS = get_excluded_urls(_CONFIG_PREFIX)
+_SPAN_KEY = "opentelemetry-instrumentor-aiohttp-server.span_key"
+
+
+def _span_name(request: BaseRequest) -> str:
+    return f"HTTP {request.method.upper()}"
+
+
+def _is_suppressed(excluded_urls: ExcludeList, request: BaseRequest) -> bool:
+    return context.get_value(
+        "suppress_instrumentation"
+    ) or excluded_urls.url_disabled(str(request.url))
+
+
+def _collect_request_attributes(
+    request: BaseRequest, url_filter: t.Callable[[str], str]
+) -> t.Dict[str, str]:
+    url = str(request.url)
+    if url_filter:
+        url = url_filter(url)
+
+    return {
+        SpanAttributes.HTTP_METHOD: request.method.upper(),
+        SpanAttributes.HTTP_URL: url,
+        SpanAttributes.HTTP_FLAVOR: ".".join(map(str, request.version)),
+    }
+
+
+def _set_span_status(span: trace.Span, response: StreamResponse):
+    if not span.is_recording():
+        return
+
+    reason = None
+    try:
+        http_status = int(response.status)
+    except ValueError:
+        http_status = response.status
+        reason = f"Unable to parse status code: {http_status!r}"
+    except AttributeError:
+        reason = "Response without a status code"
+
+    if reason:
+        span.set_status(trace.Status(trace.StatusCode.ERROR), reason=reason)
```

Comment:
I believe this is wrong. It would be something like
```suggestion
        span.set_status(trace.Status(trace.StatusCode.ERROR, description=reason))
```